src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1644,29 +1767,57 @@ private static int LocateLastFoundByte(ulong match)
                                                        0x01ul << 48) + 1;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe short LoadShort(ref byte start, IntPtr offset)
-            => Unsafe.ReadUnaligned<short>(ref Unsafe.AddByteOffset(ref start, offset));
+        private static ushort LoadUShort(ref byte start)
+            => Unsafe.ReadUnaligned<ushort>(ref start);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ushort LoadUShort(ref byte start, nuint offset)
+            => Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref start, (IntPtr)offset));
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe int LoadInt(ref byte start, IntPtr offset)
-            => Unsafe.ReadUnaligned<int>(ref Unsafe.AddByteOffset(ref start, offset));
+        private static uint LoadUInt(ref byte start)
+            => Unsafe.ReadUnaligned<uint>(ref start);
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe UIntPtr LoadUIntPtr(ref byte start, IntPtr offset)
+        private static uint LoadUInt(ref byte start, nuint offset)
+            => Unsafe.ReadUnaligned<uint>(ref Unsafe.AddByteOffset(ref start, (IntPtr)offset));
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static nuint LoadNUInt(ref byte start)
+            => Unsafe.ReadUnaligned<nuint>(ref start);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static nuint LoadNUInt(ref byte start, nuint offset)
+            => Unsafe.ReadUnaligned<nuint>(ref Unsafe.AddByteOffset(ref start, (IntPtr)offset));
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static UIntPtr LoadUIntPtr(ref byte start, IntPtr offset)
             => Unsafe.ReadUnaligned<UIntPtr>(ref Unsafe.AddByteOffset(ref start, offset));
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe Vector<byte> LoadVector(ref byte start, IntPtr offset)
+        private static Vector<byte> LoadVector(ref byte start, IntPtr offset)
             => Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref start, offset));
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe Vector128<byte> LoadVector128(ref byte start, IntPtr offset)
+        private static Vector<byte> LoadVector(ref byte start, nint offset)
+            => Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref start, (IntPtr)offset));
```

Comment:
Once this method is updated to use `nuint` instead of `nint`, use existing internal overload `Unsafe.AddByteOffset<T>(ref T, nuint)` instead.

Reply:
Done